### PR TITLE
feat(redshift): Tables can include comments

### LINKED ISSUE #22682: @aws-cdk/aws-redshift-alpha: Add comments for redshift tables construct
### Describe the feature

Currently we have this construct:

https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-redshift-alpha.Table.html

In order to be able to fully utilize CI/CD for tables and generate version controlled documentation, we need to be able to add table descriptions through code. A simple `description` attribute like in Glue tables should be enough:

https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-glue.Table.html

### Use Case

The current table construct allows the programmatic creation and modification of tables, however,  in real life,  comments for documentation are critical in any big data use case for automated documentation.

This addition will bring CDK one step closer to being able to fully manage a Redshift's table table structure and documentation.

### Proposed Solution

Add a `description` attribute to the table construct.

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.45

### Environment details (OS name and version, etc.)

AmazonLinux2

### PR DESCRIPTION
**Pull request by Rizxcviii**
Adds comments to tables.

closes #22682

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-redshift/lib/table.ts [hunk 1]
```diff
@@ -117,6 +117,13 @@ export interface TableProps extends DatabaseOptions {
    * @default cdk.RemovalPolicy.Retain
    */
   readonly removalPolicy?: cdk.RemovalPolicy;
+
+  /**
+     * A comment to attach to the table.
+     *
+     * @default - no comment
+     */
+  readonly comment?: string;
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/table.ts:126]:
```suggestion
  readonly tableComment?: string;
```

File: packages/@aws-cdk/aws-redshift/lib/private/handler-props.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export interface TableHandlerProps {
   readonly tableColumns: Column[];
   readonly distStyle?: TableDistStyle;
   readonly sortStyle: TableSortStyle;
+  readonly comment?: string;
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/private/handler-props.ts:23]:
```suggestion
  readonly tableComment?: string;
```

File: packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts [hunk 3]
```diff
@@ -60,6 +60,11 @@ async function createTable(
   }
 
   await executeStatement(statement, tableAndClusterProps);
+
+  if (tableAndClusterProps.comment) {
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts:64]:
```suggestion
  if (tableAndClusterProps.tableComment) {
```

**Review by comcalvi — CHANGES_REQUESTED** [review]
nice work 🙂 could we change the name from `comment` to `tableComment`, since SQL supports comments on columns as well?

File: packages/@aws-cdk/aws-redshift/lib/table.ts (hunk 1 above)
Comment (Rizxcviii) [reply packages/@aws-cdk/aws-redshift/lib/table.ts]: Done

File: packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts (hunk 3 above)
Comment (Rizxcviii) [reply packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts]: Done

File: packages/@aws-cdk/aws-redshift/lib/private/handler-props.ts (hunk 2 above)
Comment (Rizxcviii) [reply packages/@aws-cdk/aws-redshift/lib/private/handler-props.ts]: Done

**Review by comcalvi — APPROVED** [review]

**Review by comcalvi — DISMISSED** [review]

**Comment by Rizxcviii** [comment]
@mergifyio refresh

**Comment by Rizxcviii** [comment]
@Mergifyio refresh

**Comment by Rizxcviii** [comment]
@Mergifyio refresh

**Comment by Rizxcviii** [comment]
@Mergifyio refresh

**Comment by Rizxcviii** [comment]
@Mergifyio refresh

**Comment by Rizxcviii** [comment]
@Mergifyio refresh

**Comment by Rizxcviii** [comment]
@comcalvi Within my fork, I've created a personal access token with read/write access to workflows, added it as a secret to the repository, and the merge cannot happen for these issues.

> Base branch update has failed

> refusing to allow a GitHub App to create or update workflow .github/workflows/close-stale-prs.yml without workflows permission
err-code: 9AD93

**Comment by comcalvi** [comment]
very strange...not sure what's going on here

**Review by comcalvi — APPROVED** [review]

**Pull request merged**